test/models/activation_key_test.rb

```diff
@@ -107,6 +107,16 @@ def setup
       end
     end
 
+    test "should not allow generated content view to be associated with activation key" do
+      generated_cv = FactoryBot.create(:katello_content_view, :generated_for => :repository_export, :organization => @dev_view.organization)
+
+      exception = assert_raises(ActiveRecord::RecordInvalid) do
+        @dev_key.update!(content_view: generated_cv)
+      end
+
+      assert_match(/Generated content views cannot be assigned to hosts or activation keys/, exception.message)
+    end
```

Comment:
```suggestion
    test "should not allow generated content view to be associated with activation key" do
      generated_cv = FactoryBot.create(:katello_content_view, :generated_for => :repository_export, :organization => @dev_view.organization)
      generated_cve = FactoryBot.create(:katello_content_view_environment, :content_view => generated_cv, :environment => katello_environments(:library))

      exception = assert_raises(ActiveRecord::RecordInvalid) do
        @dev_key.update!(content_view_environments: [generated_cve])
      end

      assert_match(/Generated content views cannot be assigned to hosts or activation keys/, exception.message)
    end
```

Something along these lines looks necessary due to the changes in https://github.com/Katello/katello/commit/746bda76b50cbe6a321f17994dcd0cd13d97c49b#diff-d7dd2b8980210ef5f81a8ac4d2bfd2384210cf326a1b165a5b6869b0c661a241